server/src/main/java/org/eclipse/openvsx/LocalRegistryService.java

```diff
@@ -426,7 +460,21 @@ public ExtensionJson publish(InputStream content, String tokenValue) throws Erro
         eclipse.checkPublisherAgreement(token.getUser());
 
         var extVersion = extensions.publishVersion(content, token);
-        return toExtensionVersionJson(extVersion, true);
+        var json = toExtensionVersionJson(extVersion, null, true);
+        var sameVersions = repositories.findVersions(extVersion.getVersion(), extVersion.getExtension());
+        if(sameVersions.stream().anyMatch(ev -> ev.isPreRelease() != extVersion.isPreRelease())) {
+            var existingRelease = extVersion.isPreRelease() ? "stable release" : "pre-release";
+            var thisRelease = extVersion.isPreRelease() ? "pre-release" : "stable release";
+            var extension = extVersion.getExtension();
+            var semver = extVersion.getSemanticVersion();
+            var newVersion = String.join(".", String.valueOf(semver.getMajor()), String.valueOf(semver.getMinor() + 1), "0");
+
+            json.warning = "A " + existingRelease + " already exists for " +
+                    extension.getNamespace().getName() + "." + extension.getName() + "-" + extVersion.getVersion() + ".\n" +
+                    "To prevent update conflicts we recommend that this " + thisRelease + " uses " + newVersion + " as version instead.";
```

Comment:
```suggestion
                    "To prevent update conflicts, we recommend that this " + thisRelease + " uses " + newVersion + " as its version instead.";
```